### DISCUSSION TITLE
Issue with ‘standard.link’ Disappearing on Multiple Clicks

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2632] [Q&A] [answered]
### Introduction

Hello,

I am currently working on a project where I save the JSON string of graph elements and import them back to the graph for rendering. Recently, I encountered an issue with one of the graphs. When I click more than once on the link (‘standard.link’), the link disappears, and an error appears in the console indicating a NaN element in the path.

I have cross-checked to see if there is any null or NaN in the source and target of the standard link, but I found no such link. I have tried debugging in various ways but have not been able to find a proper direction to move forward.

For better understanding, I have attached a video showing how links are disappearing on the paper. I have also attached logs for console errors and an XML file which contains the JSON string of graph elements. There are many custom elements in the string, so you might not be able to directly render the graph. However, you can see the elements of standard links in the string which might be helpful for debugging.

If any further information is required, please let me know. Any help or guidance would be greatly appreciated.

files:
[logs and videos.zip](https://github.com/clientIO/joint/files/15128563/logs.and.videos.zip)

Thank you.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
This is a wild guess, but was the value of `paper.options.gridSize`? Is it a number?

If yes, please provide steps to reproduce. It's not happening in any of our current examples, correct?

**Reply by VADDIJASWANT** [reply]
Hello @kumilingus,

I apologize for the delayed response. Initially, I was unable to reproduce the issue in my local environment. However, after considering your query regarding the grid size, I delved a bit deeper into that aspect and indeed discovered the problem.

The issue seems to arise when the gridSize is set to 0. Interestingly, I noticed in the online JointJS Kitchen Sink example that the minimum gridSize provided is 1. When I adjusted the gridSize accordingly, the functionality appears to be mostly working fine now.

Small Doubt is the MinGrid Size intentionally not set to 0 in the kitchen sink application? Thank you.

**Reply by kumilingus** [reply, marked as answer]
The `gridSize` must be `> 0` currently. So it is set to `1` for a reason.
